components/content/Icon/Icon.stories.ts

```diff
@@ -0,0 +1,254 @@
+import { StoryObj } from '@storybook/web-components';
+import { html } from 'lit';
+
+import './Icon';
+
+const description = `
+Wraps an svg icon to provide alignment and sizing.
+
+Icons must have the appropriate fill or stroke set to currentColor.
+
+Icons are hidden from screen readers by default and should be accompanied by text. If an icon must be used on its own, add an aria-label to the icon element.
```

Comment:
The aria-label would need to go on the SVG not the diamond-icon. I think the custom element will have a generic role https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-label#associated_roles

Reply:
It looks like aria-label should be on the interactive element (the button or link). I've updated the docs

> Note: While aria-label can be used on any element that can have an accessible name, in practice however, it is supported only on interactive elements, [widgets](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles#2._widget_roles), [landmarks](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles#3._landmark_roles), images, and iframes.

Reply:
SVG has an image role so we need to support aria label on the SVG to handle adding a label to an SVG when not placed in an interactive element.

One example use case I can think of for that is when the twitter x icon is used in a string of text.

In the angular lib we use it accepts the label on the component then passes it down to the SVG, or we could detect presence of aria-label on the SVG and not apply aria-hidden to the diamon-icon